src/commands/play.rs

```diff
@@ -102,6 +118,42 @@ pub async fn execute_play(
     Ok(())
 }
 
+async fn calculate_time_until_play(queue: &[TrackHandle], flag: &PlayFlag) -> Option<Duration> {
+    if !queue.is_empty() {
+        let top_track = queue.first().expect("Could not fetch playing song");
+
+        let top_track_position = top_track
+            .get_info()
+            .await
+            .expect("Could not get playing track info")
+            .position;
+
+        let top_track_duration = top_track
+            .metadata()
+            .duration
+            .expect("Could not fetch duration of top track");
+
+        let mut estimated_time = match flag {
+            PlayFlag::DEFAULT => queue[1..queue.len() - 1]
+                .iter()
+                .fold(Duration::ZERO, |acc, x| {
+                    acc + x
+                        .metadata()
+                        .duration
+                        .expect("Could not fetch duration of track")
+                }),
+            PlayFlag::PLAYTOP => Duration::ZERO,
+        };
+
+        // Add the remaining top track
+        estimated_time += top_track_duration - top_track_position;
```

Comment:
So `top_track_position` is not like a position index (0 1 2 or 3) right? I think its something else and perhaps rename it.

Reply:
Good point. In this context, position is the elapsed time for a track (that's what Songbird calls it), but it does make you think it is a index. I'll rename it to something more clear.